internal/notifier/factory.go

```diff
@@ -64,42 +64,92 @@ type notifierMap map[string]factoryFunc
 type factoryFunc func(opts notifierOptions) (Interface, error)
 
 type notifierOptions struct {
-	URL         string
-	ProxyURL    string
-	Username    string
-	Channel     string
-	Token       string
-	Headers     map[string]string
-	CertPool    *x509.CertPool
-	Password    string
-	ProviderUID string
+	URL          string
+	ProxyURL     string
+	Username     string
+	Channel      string
+	Token        string
+	Headers      map[string]string
+	CertPool     *x509.CertPool
+	Password     string
+	CommitStatus string
 }
 
 type Factory struct {
 	notifierOptions
 }
 
-func NewFactory(url string,
-	proxy string,
-	username string,
-	channel string,
-	token string,
-	headers map[string]string,
-	certPool *x509.CertPool,
-	password string,
-	providerUID string) *Factory {
+// NotifierOption represents a functional option for configuring a notifier.
+type NotifierOption func(*notifierOptions)
```

Comment:
I think I'd call this type just `Option` since this is the most important option type of the package and `notifier.NotifierOption` "stutters" in Go idiom